### PR TITLE
cleanup: moves raster-merging logic into raster_merger.py

### PR DESCRIPTION
**Pull request by strangeties**

### CODE REVIEW

File: src/planscape/forsys/raster_merger.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                "invalid raster has %d bands (expected 1)" %
+                (len(raster.bands)))
+        if not np.isnan(raster.bands[0].nodata_value):
+            raise Exception(
+                "invalid raster has nodata value, %s (expected np.nan)" %
+                (raster.bands[0].nodata_value))
+
+    def _ensure_raster_is_compatible_with_merged_raster(
+            self, raster: GDALRaster) -> None:
+        scale = self.merged_raster.scale
```
Comment (riecke) [inline src/planscape/forsys/raster_merger.py:48]: Maybe add a defensive check before this line?

  if self.merged_raster is None:
    return

**Review by riecke — APPROVED** [review]
Really nice, thorough testing here!

**Comment by elsiel23** [comment]
QQ, I think rasterio has a merge function as well. Are there any limitations that make it difficult for us to use here?

File: src/planscape/forsys/raster_merger.py (hunk 1 above)
Comment (strangeties) [reply src/planscape/forsys/raster_merger.py:53]: sure - done.
I've added the check to all the protected methods that refer to self.merged_raster.

**Comment by strangeties** [comment]
Thanks for reviewing!

`QQ, I think rasterio has a merge function as well. Are there any limitations that make it difficult for us to use here?`
I looked briefly into gdal_merge.py and rasterio, but was unsure how to work with the input formats. A lot of examples involved reading a raster that's saved to disk while others suggested calling shell script from python. The functions in https://docs.djangoproject.com/en/4.1/ref/contrib/gis/gdal/#raster-data-objects were the most immediately accessible so I coded around them.

**Pull request merged**